CefSharp.Core/Internals/ClientAdapter.cpp

```diff
@@ -285,15 +315,15 @@ namespace CefSharp
                     IBrowser^ entry;
                     if (_popupBrowsers->TryGetValue(browserId, entry))
                     {
-                        auto popupHandler = _browserControl->PopupHandler;
-                        auto frameWrapper = gcnew CefFrameWrapper(frame, _browserAdapter);
-                        popupHandler->OnFrameLoadEnd(_browserControl, entry, gcnew FrameLoadEndEventArgs(frameWrapper, httpStatusCode));
+                        CefFrameWrapper frameWrapper(frame, _browserAdapter);
+                        popupHandler->OnFrameLoadEnd(_browserControl, entry, gcnew FrameLoadEndEventArgs(%frameWrapper, httpStatusCode));
                     }
                 }
             }
             else
             {
-                _browserControl->OnFrameLoadEnd(gcnew CefFrameWrapper(frame, _browserAdapter), httpStatusCode);
+                CefFrameWrapper frameWrapper(frame, _browserAdapter);
```

Comment:
Interesting. I take it the lifespan is only valid for the life of the call? Should this be the standard approach for all those that we would otherwise immediately dispose of?